src/Explorer/Explorer.tsx

```diff
@@ -292,32 +292,87 @@ export default class Explorer {
     const baseUrl = `vscode://ms-azuretools.vscode-cosmosdb?resourceId=${resourceId}`;
     const vscodeUrl = activeTab ? `${baseUrl}&database=${database}&container=${container}` : baseUrl;
 
-    const openVSCodeDialogProps: DialogProps = {
-      linkProps: {
-        linkText: "Download Visual Studio Code",
-        linkUrl: "https://code.visualstudio.com/download",
-      },
-      isModal: true,
-      title: `Open your Azure Cosmos DB account in Visual Studio Code`,
-      subText: `Please ensure Visual Studio Code is installed on your device.
-      If you don't have it installed, please download it from the link below.`,
-      primaryButtonText: "Open in VS Code",
-      secondaryButtonText: "Cancel",
+    // Detect if VS Code is installed
+    const detectVSCode = () => {
+      return new Promise<boolean>((resolve) => {
+        // Create hidden iframe to detect protocol handler
+        const iframe = document.createElement("iframe");
+        iframe.style.display = "none";
+        document.body.appendChild(iframe);
+
+        const timeoutId = setTimeout(() => {
+          if (document.body.contains(iframe)) {
+            document.body.removeChild(iframe);
+          }
+          resolve(false);
+        }, 1000);
+
+        try {
+          // Listen for blur event which might indicate app was launched
+          const onBlur = () => {
+            clearTimeout(timeoutId);
+            window.removeEventListener("blur", onBlur);
+            if (document.body.contains(iframe)) {
+              document.body.removeChild(iframe);
+            }
+            // Window lost focus, likely because VS Code opened
+            resolve(true);
+          };
+          window.addEventListener("blur", onBlur, { once: true });
+          // Use a test URL that won't open a specific resource but will check if the extension is installed
+          iframe.src = "vscode://ms-azuretools.vscode-cosmosdb?test=1";
```

Comment:
This can't work, VS Code will load the extension and pass the "test=1" param to our handler which will fail of course. This results in an additional VS Code prompt:
<img width="533" alt="image" src="https://github.com/user-attachments/assets/b927dc6f-e108-4604-a5f5-2f4058bb9ced" />
Which then fails:
<img width="460" alt="image" src="https://github.com/user-attachments/assets/3b41500c-3c40-4438-b357-ae978e42d1ac" />
Even if we would handle the test parameter and ignore it, VS Code will still prompt twice unless the user has opted in to not being asked again to handle external URLs.